onnxruntime/python/onnxruntime_cuda_temp_env.py

```diff
@@ -0,0 +1,40 @@
+import ctypes
+import os
+import platform
+
+
+def get_nvidia_lib_paths():
+    import site
+
+    # Get the site-packages path where nvidia packages are installed
+    site_packages_path = site.getsitepackages()[0]
+    nvidia_path = os.path.join(site_packages_path, "nvidia")
+    # Collect all directories under site-packages/nvidia that contain .dll files (for Windows)
+    lib_paths = []
+    if platform.system() == "Windows":  # Windows
+        for root, _, files in os.walk(nvidia_path):
+            if any(file.endswith(".dll") for file in files):
+                lib_paths.append(root)
+    elif platform.system() == "Linux":
+        import re
+
+        pattern = re.compile(r"\.so(\.\d+)?$")
+        for root, _, files in os.walk(nvidia_path):
+            for file in files:
+                if pattern.search(file):
+                    lib_paths.append(root)
+    else:
+        pass
+    return lib_paths
+
+
+def load_nvidia_libs():
+    cuda_lib_paths = get_nvidia_lib_paths()
+    if platform.system() == "Windows":
+        for path in cuda_lib_paths:
+            os.add_dll_directory(path)
+    elif platform.system() == "Linux":
+        for path in cuda_lib_paths:
+            _ = ctypes.CDLL(path)
```

Comment:
There is no need to load every DLL. For example cudnn 9 has many DLL files, we only need load the main DLL like
```
        cuda_libs: _Dict[str, str] = {
            "cublas": "libcublas.so.*[0-9]",
            "cudnn": "libcudnn.so.*[0-9]",
            "cuda_runtime": "libcudart.so.*[0-9]",
            "cuda_nvrtc": "libnvrtc.so.*[0-9]", // optional?            
            "cufft": "libcufft.so.*[0-9]",            // optional? 
            "curand": "libcurand.so.*[0-9]",    // optional?
            "nvjitlink": "libnvJitLink.so.*[0-9]", // optional?
        }
```

Reply:
Do you have the list of main dll and so we need to load?

Reply:
Now it should only load 
```
        cuda_libs = (
            "cublas.dll",
            "cudnn.dll",
            "cudart.dll",
            "nvrtc.dll",
            "cufft.dll",
            "curand.dll",
            "nvJitLink.dll",
        )
        ```
on windows
```

        cuda_libs = (
            "libcublas.so",
            "libcudnn.so",
            "libcudart.so",
            "libnvrtc.so",
            "libcufft.so",
            "libcurand.so",
            "libnvJitLink.so",
        )
```
on Linux where `.so` could have and `.n` suffix where `n>=0`